### DISCUSSION TITLE
How do I apply outline only when a data attribute exists (is true)?

### DISCUSSION THREAD

**Discussion by daniellevinson** [tailwindlabs/tailwindcss#12032] [Help] [answered]
If I remove the outline by default with `outline-none`, `outline-0` or `outline-transparent`, how do I apply back the default color/width when `data-focus-visible` is `true`?
There's no `:not` selector or `outline-default` or setting only `outline-style` to `none` as far as I'm aware.
Also I can't set `data-focus-visible` to `false` on the element - using a 3rd party library that sets the attributes for me.

**Comment by wongjn** [comment, marked as answer]
To set `outline-style: none`, you could use an [arbitrary property class](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-properties) like `[outline-style:none]`. For a `:not()` selector, you could consider using an [arbitrary variant](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-variants) like `[&:not([data-focus-visible=true])]:outline-none`.

**Reply by daniellevinson** [reply]
Thanks!
